test/e2e/persistent_volumes-gce.go

```diff
@@ -107,11 +107,6 @@ var _ = framework.KubeDescribe("PersistentVolumes:GCEPD [Volume]", func() {
 			framework.DeletePodWithWait(f, c, clientPod)
 			framework.PVPVCCleanup(c, ns, pv, pvc)
 			clientPod, pv, pvc, node = nil, nil, nil, ""
-		}
-	})
-
-	AddCleanupAction(func() {
-		if len(diskName) > 0 {
```

Comment:
1.  Added check
2. diskName is assigned "" if there is an error in disk creation.